libraries/AP_Camera/examples/tracking.py

```diff
@@ -0,0 +1,161 @@
+import cv2
+import threading
+from pymavlink import mavutil
+import math 
+from ultralytics import YOLO
+
+master = mavutil.mavlink_connection('127.0.0.1:14560')
+master.wait_heartbeat()
+print("Heartbeat from the system (system %u component %u)" % 
+     (master.target_system, master.target_component))
+def send_gimbal_manager_pitch_yaw_angles(target_system, target_component, flags, gimbal_device_id, pitch, yaw, pitch_rate, yaw_rate):
+    """
+    Send gimbal manager pitch and yaw angles message.
+
+    :param target_system: System ID
+    :param target_component: Component ID
+    :param flags: High level gimbal manager flags to use
+    :param gimbal_device_id: Component ID of gimbal device to address (or 1-6 for non-MAVLink gimbal), 0 for all gimbal device components
+    :param pitch: Pitch angle (positive: up, negative: down, NaN to be ignored)
+    :param yaw: Yaw angle (positive: to the right, negative: to the left, NaN to be ignored)
+    :param pitch_rate: Pitch angular rate (positive: up, negative: down, NaN to be ignored)
+    :param yaw_rate: Yaw angular rate (positive: to the right, negative: to the left, NaN to be ignored)
+    """
+    msg = master.mav.gimbal_manager_set_pitchyaw_encode(
+        target_system,
+        target_component,
+        flags,
+        gimbal_device_id,
+        pitch,
+        yaw,
+        pitch_rate,
+        yaw_rate
+    )
+    master.mav.send(msg)
+    #print("Gimbal manager pitch and yaw angles message sent.")
+send_gimbal_manager_pitch_yaw_angles(0, 0, 0, 0, 0, 0, float("NaN"),float("NaN"))
+def send_command():
+    global centre_x, center_y
+    while(1):
+        centre_x_copy = int(center_x)
+        centre_y_copy = int(center_y)
+        if ((centre_x_copy == 0) and (centre_y_copy ==0)):
+            diff_x = 0
+            diff_y = 0
+        else:
+            diff_x = (centre_x_copy - (640/2))/2
+            diff_y = -(centre_y_copy - (480/2))/2
+        #print(diff_x,diff_y)
+
+        send_gimbal_manager_pitch_yaw_angles(0, 0, 0, 0, float("NaN"), float("NaN"), math.radians(diff_y), math.radians(diff_x))
+
+#########################################################
+# Initialize the VideoCapture object
+
+# gst-launch-1.0 -v udpsrc port=5600 caps='application/x-rtp, media=(string)video, clock-rate=(int)90000, encoding-name=(string)H264' ! rtph264depay ! avdec_h264 ! videoconvert ! autovideosink sync=false
+
+gst_pipeline = (
+    "udpsrc port=5600 ! application/x-rtp,encoding-name=H264,payload=96 ! "
+    "rtph264depay ! h264parse ! queue ! avdec_h264 ! videoconvert ! appsink"
+)
+
+cap = cv2.VideoCapture(gst_pipeline, cv2.CAP_GSTREAMER)
```

Comment:
The default versions of `opencv-python` and `opencv-contrib-python` do not have gstreamer support enabled. A custom build of the Python packages from source is required for this (at least this is the case on macOS). To reduce the complexity it may be easier to use the gstreamer Python bindings to capture the video stream and then make this available to opencv. The BlueROV docs have a guide for this: https://www.ardusub.com/developers/opencv.html.